Multiple hibernate versions

Currently we use  two `hibernate` versions (`5.0` and `5.2`) in our zally server project: 

``
./gradlew dependencyInsight --dependency hibernate
``

```
org.hibernate:hibernate-core:5.2.9.Final (selected by rule)
\--- compile

org.hibernate:hibernate-core:5.0.12.Final -> 5.2.9.Final
+--- org.hibernate:hibernate-entitymanager:5.0.12.Final
|    +--- org.springframework.boot:spring-boot-starter-data-jpa:1.5.7.RELEASE
|    |    \--- compile
|    \--- org.jadira.usertype:usertype.core:6.0.1.GA
|         \--- compile
\--- org.springframework.boot:spring-boot-starter-data-jpa:1.5.7.RELEASE (*)

org.hibernate:hibernate-entitymanager:5.0.12.Final (selected by rule)
\--- org.springframework.boot:spring-boot-starter-data-jpa:1.5.7.RELEASE
     \--- compile

org.hibernate:hibernate-entitymanager:5.2.1.Final -> 5.0.12.Final
\--- org.jadira.usertype:usertype.core:6.0.1.GA
     \--- compile

org.hibernate.javax.persistence:hibernate-jpa-2.1-api:1.0.0.Final
+--- org.hibernate:hibernate-core:5.2.9.Final
|    +--- compile
|    +--- org.springframework.boot:spring-boot-starter-data-jpa:1.5.7.RELEASE
|    |    \--- compile
|    \--- org.hibernate:hibernate-entitymanager:5.0.12.Final
|         +--- org.springframework.boot:spring-boot-starter-data-jpa:1.5.7.RELEASE (*)
|         \--- org.jadira.usertype:usertype.core:6.0.1.GA
|              \--- compile
\--- org.hibernate:hibernate-entitymanager:5.0.12.Final (*)
```

This can lead to unintentional behaviour: I tried to update the gradle version to ``4.3`` and struggled with the following exception during the test execution (I assume that the gradle version update changed the selection of the transitive dependency): 

```
    Caused by: java.lang.NoSuchMethodError: org.hibernate.engine.spi.SessionFactoryImplementor.getProperties()Ljava/util/Properties;
        at org.hibernate.jpa.internal.EntityManagerFactoryImpl.<init>(EntityManagerFactoryImpl.java:124) ~[hibernate-entitymanager-5.0.12.Final.jar:5.0.12.Final]
        at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:890) ~[hibernate-entitymanager-5.0.12.Final.jar:5.0.12.Final]
        at org.springframework.orm.jpa.vendor.SpringHibernateJpaPersistenceProvider.createContainerEntityManagerFactory(SpringHibernateJpaPersistenceProvider.java:60) ~[spring-orm-4.3.11.RELEASE.jar:4.3.11.RELEASE]
```

Proposed solutions: 

1. Downgrade Jadira and use hibernate 5.0
Jadira version `6.0.1.GA` is not compatible with hibernate `5.0`.
Use `Jadira` version `5.0.0.GA` and remove `hibernate-core` dependency.  

2. Only use hibernate 5.2 
use `hibernate-core:5.2.1.Final ` and  exclude `hibernate-entitymanager` as it is deprecated.

--- a/server/build.gradle
+++ b/server/build.gradle
@@ -57,12 +57,16 @@ dependencies {
     compile("org.springframework.boot:spring-boot-starter-web")
     compile("org.springframework.boot:spring-boot-starter-undertow")
     compile("org.springframework.boot:spring-boot-starter-actuator")
-    compile('org.springframework.boot:spring-boot-starter-data-jpa')
+    compile('org.springframework.boot:spring-boot-starter-data-jpa') {
+        exclude group: "org.hibernate", module: "hibernate-entitymanager"
+    }
     compile('org.flywaydb:flyway-core:4.1.2')
     compile('org.hsqldb:hsqldb:2.3.3')
     compile('org.postgresql:postgresql:42.1.1')
     compile('org.hibernate:hibernate-core:5.2.9.Final')
-    compile('org.jadira.usertype:usertype.core:6.0.1.GA')
+    compile('org.jadira.usertype:usertype.core:6.0.1.GA') {
+        exclude group: "org.hibernate", module: "hibernate-entitymanager"
+    }
     compile("com.fasterxml.jackson.module:jackson-module-parameter-names")
     compile("com.fasterxml.jackson.datatype:jackson-datatype-jsr310")
     compile("com.fasterxml.jackson.datatype:jackson-datatype-jdk8")

